mempool/mempool.go

```diff
@@ -299,7 +308,8 @@ func (mem *Mempool) resCbNormal(req *abci.Request, res *abci.Response) {
 	switch r := res.Value.(type) {
 	case *abci.Response_CheckTx:
 		tx := req.GetCheckTx().Tx
-		if r.CheckTx.Code == abci.CodeTypeOK {
+		postCheckFilterPasses := (mem.postCheckFilter == nil) || mem.postCheckFilter(tx, r.CheckTx)
```

Comment:
It's repeated, might be worth to refactor into a function which can be called inside of the if statement itself.